adapters/repos/db/index.go

```diff
@@ -1643,10 +1643,18 @@ func (i *Index) addNewShard(ctx context.Context,
 	return nil
 }
 
-func (i *Index) validateMultiTenancy(tenantKey string) error {
+func (i *Index) validateMultiTenancy(tenantKey string, object *models.Object) error {
```

Comment:
It would be useful to return the tenantKey along the error. This would eliminate the need to parse the key separately, resulting in improved efficiency